app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,57 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
+  end
+
+  def destroy
+  end
+end
+
+def create
+  user = User.find_by(username: params[:username])
+
+  if user
+    # Success
+  else
+    flash.now[:alert] = 'Invalid username or password.'
+    render :new
+  end
+end
+
+def create
+  user = User.find_by(username: params[:username])
+
+  if user
+    if user.authenticate(params[:password])
+      # Success
+
+    else
+      flash.now[:alert] = 'Invalid username or password.'
+      render :new
+    end
+  else
+    flash.now[:alert] = 'Invalid username or password.'
+    render :new
+  end
+end
+
+def create
+  user = User.find_by(username: params[:username])
+
+  if user
+    if user.authenticate(params[:password])
+      # Let's set user_id and also role
+      session[:user_id] = user.id
+      session[:role] = user.role
+      redirect_to root_path
+    else
+      flash.now[:alert] = 'Invalid username or password.'
+      render :new
+    end
+  else
+    flash.now[:alert] = 'Invalid username or password.'
+    render :new
+  end
+end
```

Comment:
```suggestion
class SessionsController < ApplicationController
  def new
  end

  
  def create
    user = User.find_by(username: params[:username])
  
    if user
      if user.authenticate(params[:password])
        # Success
  
      else
        flash.now[:alert] = 'Invalid username or password.'
        render :new
      end
    else
      flash.now[:alert] = 'Invalid username or password.'
      render :new
    end
  end

  def destroy
  end
end
```


multiple create method, also not in the session controller class. apply and modify as needed